external-crates/move/crates/move-package/src/resolution/dependency_graph.rs

```diff
@@ -1209,11 +1209,19 @@ impl DependencyGraph {
         Ok(graph)
     }
 
-    /// Serializes this dependency graph into a lock file and return it.
+    /// Serializes this dependency graph into a lock file and returns it.
     ///
     /// This operation fails, writing nothing, if the graph contains a cycle, and can fail with an
     /// undefined output if it cannot be represented in a TOML file.
-    pub fn write_to_lock(&self, install_dir: PathBuf) -> Result<LockFile> {
+    ///
+    /// `install_dir` is a working directory to create a lock file with dependency graph info.
+    /// `lock_path` is an optional parameter: if it is set, and exists, this `Move.lock` will be
+    /// updated with the dependency graph content. If not, the lock file is created from scratch.
+    pub fn write_to_lock(
+        &self,
+        install_dir: PathBuf,
+        lock_path: Option<PathBuf>,
+    ) -> Result<LockFile> {
```

Comment:
This can also be part of the doc comment, so feel free to suggest putting it in here, but to be clear on how this function is changed:

```
    /// The optional parameter `lock_path` applies to two different uses of this function:                                                                                                                                                                              
    ///                                                                                                                                                                                                                                                     
    /// (1) The computation of a digest of the root package's dependencies. Here we only                                                                                                                                                                    
    /// care that the returned lock file contents reflect the dependency graph, so the initial lock file                                                                                                                                                    
    /// is created from scratch (`lock_path` is None). It used by callers for a transient operation: the                                                                                                                                                    
    /// lock files for dependencies are thrown away after the digest is computed.                                                                                                                                                                           
    ///                                                                                                                                                                                                                                                     
    /// (2) Allowing a root package's `Move.lock` to be updated with graph info, while preserving existing                                                                                                                                                  
    /// `Move.lock` content. When this is the intent of the caller, `lock_path` may be set to `Some`                                                                                                                                                        
    /// existing `Move.lock` file. It is a persistent operation in this context: the `Move.lock` file is                                                                                                                                                    
    /// updated in the root package's directory.   
```